compression/lempel_ziv_factorization.py

```diff
@@ -0,0 +1,26 @@
+
+from compression import lpf
+
+
+def naive(w, n):
+  LZ = []
+  i = 1
+  while i <= n:
+    matched = [w[i:(i + k)]
+           for k in range(1, n) if w.index(w[i:(i + k)]) < i]
+    longest = (matched or [w[i]])[-1]
+    LZ += [len(longest)]
+    i += len(longest)
+
+  return [-1] + LZ
```

Comment:
Please take a look at lempel_ziv.py. First, this code should go there regardless of any other considerations. Second, it has a different output and I'm open to listening about the advantages and disadvantages of both formats - but I guess the format adopted in lempel_ziv.py is necessary to perform inverse (recovering text from its LZ array). So I strongly suggest you to migrate to that format, unless convinced otherwise.

Reply:
I don't understand what functions in `lempel_ziv.py` return. It looked to me that it is something different that the factorization. That is why I created a new file. I could not connect `lempel_ziv.py` with the outputs that I got. For instance, `lempel_ziv.py` returned less factors (a smaller number of them) than my algorithm, for the same string.

As for the format I chose here - it is the same as in LPF array - the lengths of consequtive strings. It made sense for me because LPF is permutation of LCP, and LZ is almost the LPF itself, but filtered and with zeroes replaced by ones.
Also, this format is close to what is used in the paper - in the paper there are indexes of starting positions of factors. But I thought that it is just a complication to transform it that way.

Reply:
I used the format from here: https://en.wikipedia.org/wiki/LZ77_and_LZ78#Pseudocode - i.e. (offset of the start, lenght [I guess LPF], following char).

Your format make sense for LPF, but I suppose LZ transformations should be reversible. I guess you can also use only 2, not 3 items in each tuple as in https://arxiv.org/pdf/1212.2952.pdf but since this is less readable and the repo is for didactic purposes, I find the one above the most convenient.

Reply:
Ack, thank you

Reply:
Actually, sorry - link is to the actual compression algorithm, which is different than fatorization.
So, should there still be single file for the two algorithms?

(I will try to adopt the format either way)

Reply:
Sorry, I forgot to add the exact reference inthe paper: on page 3 there's zzzzzipzip - > [(z, 0),(1, 4),(i, 0),(p, 0),(5, 3)] factorization

Reply:
I understand now, thank you